pkg/config/expression.go

```diff
@@ -544,20 +559,19 @@ func parseBpLit(s string) (cty.Value, error) {
 // The shortest expression is returned. E.g:
 // "var.hi) $(var.there)" -> "var.hi"
 // "try(var.this) + one(var.time)) tail" -> "try(var.this) + one(var.time)"
-func greedyParseHcl(s string) (Expression, string, error) {
+func greedyParseHcl(s string) (string, string, error) {
 	err := errors.New("no closing parenthesis")
 	for i := 0; i < len(s); i++ {
 		if s[i] != ')' {
 			continue
 		}
 		_, diag := hclsyntax.ParseExpression([]byte(s[:i]), "", hcl.Pos{})
 		if !diag.HasErrors() { // found an expression
-			exp, err := BlueprintExpressionLiteralToExpression(s[:i])
-			return exp, s[i+1:], err
+			return s[:i], s[i+1:], nil
 		}
 		err = diag // save error, try to find another closing bracket
 	}
-	return nil, s, err
+	return "", s, err
```

Comment:
Why this is changed from nil to empty string ?

Reply:
Because signature have changed:
```go
- func greedyParseHcl(s string) (Expression, string, error) {
+ func greedyParseHcl(s string) (string, string, error) {
```